src/hyperion/external_interaction/nexus/write_nexus.py

```diff
@@ -13,68 +13,51 @@
 from nexgen.nxs_utils import Attenuator, Beam, Detector, Goniometer, Source
 from nexgen.nxs_write.nxmx_writer import NXmxFileWriter
 from numpy.typing import DTypeLike
+from scanspec.core import AxesPoints
 
 from hyperion.external_interaction.nexus.nexus_utils import (
     create_detector_parameters,
     create_goniometer_axes,
     get_start_and_predicted_end_time,
 )
-from hyperion.parameters.internal_parameters import (
-    HyperionParameters,
-    InternalParameters,
-)
+from hyperion.parameters.gridscan import ThreeDGridScan
+from hyperion.parameters.rotation import RotationScan
 
 
 class NexusWriter:
     def __init__(
         self,
-        parameters: InternalParameters,
-        scan_points: dict,
+        parameters: ThreeDGridScan | RotationScan,
```

Comment:
Could: Better to depend on the lowest class required but in this case is something which inherits from both `DiffractionExperimentWithSample` and `OptionalGonioAngleStarts`. So we would have to make a new type for it. I think it's work we may end up having to do but it's fine for now

Reply:
In fact I think the better way to solve this is probably for this if for the caller to fill in `omega_start_deg` and `chi_start_deg` then take a `DiffractionExperimentWithSample` set of parameters. Could we either do that, argue about it or spin into a new ticket to argue about it. Up to you

Reply:
sure, I think that's a good solution